client/src/components/Applications/AppLayout/AppLayout.test.js

```diff
@@ -25,7 +44,26 @@ function renderAppsHeader(store, appId) {
     </Provider>
   );
 }
-
+it('displays an error if no storage systems are enabled', async () => {
```

Comment:
Check out `client/src/components/DataFiles/DataFilesModals/tests/DataFilesPreviewModal.test.js` as it has two typical tests to check for a spinner (when loading) and an error message if error.

The key is to construct a mock store for those two scenarios.  The mock store probably just needs to contain `apps`.  So something along the lines of what is done in client/src/components/Applications/AppBrowser/AppBrowser.test.js but with things slightly altered.  You could use the fixture and edit the `error` parameter of that object in your test.  Lets meetup to discuss.